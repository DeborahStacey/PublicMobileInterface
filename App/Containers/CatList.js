import React from 'react'
import { ScrollView, Text, Image, View } from 'react-native'
import { Images } from '../Themes'
import CatCard from '../Components/CatCard'
// import { Actions as NavigationActions } from 'react-native-router-flux'
import {create} from 'apisauce'

// Styles
import styles from './Styles/PresentationScreenStyle'

const db = create({
  baseURL: 'https://cat.ddns.net/Backend/api.php',
  headers: {'Content-Type': 'application/json'}
})

export default class CatList extends React.Component {

  constructor (props) {
    super(props)
    this.state = {
      catsData: null
    }
  }

  componentWillMount () {
    // creating a var for this inside of the promisses
    var that = this

    // The 1 here is to indicate cat for the animal type
    db.get('/pet/pets')
    .then(function (response) {
      var array = response.data
      that.setState({
        catsData: array
      })
    })
    .catch((error) => window.alert(error))
  }

  cardClicked (event) {
    window.alert('this cats id is: ' + event)
  }

  render () {
    var catsData = this.state.catsData

    var catCards = []
    if (catsData != null) {
      for (var i = 0; i < catsData.personal.length; i++) {
        catCards.push(<CatCard basicCatInfo={catsData.personal[i]} onClick={this.cardClicked.bind(this)} key={i} />)
      }
    }

    return (
      <View style={styles.mainContainer}>
        <Image source={Images.background} style={styles.backgroundImage} resizeMode='stretch' />
        <ScrollView style={styles.container}>

          <View style={styles.messageBox}>
            <View style={styles.messageBoxContents}>

              <Text style={styles.sectionText}>My Cats</Text>
              {catCards}

            </View>
          </View>

        </ScrollView>
      </View>
    )
  }
}
