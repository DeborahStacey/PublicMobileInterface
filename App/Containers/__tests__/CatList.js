import 'react-native'
import React from 'react'
import CatList from '../CatList'
import renderer from 'react-test-renderer'

it('renders correctly', () => {
  const tree = renderer.create(
    <CatList />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
