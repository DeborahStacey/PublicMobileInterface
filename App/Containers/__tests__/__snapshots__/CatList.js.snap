exports[`test renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F1F5FF",
      "flex": 1,
      "marginTop": 64,
    }
  }>
  <ScrollView
    style={
      Object {
        "flex": 1,
        "paddingTop": 10,
      }
    }>
    <View
      style={undefined}>
      <View
        style={undefined}>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#2C457F",
              "fontWeight": "bold",
              "marginVertical": 5,
              "textAlign": "center",
            }
          }>
          You don\'t have any cats yet!
        </Text>
      </View>
    </View>
  </ScrollView>
</View>
`;
